### ISSUE TITLE
Suppressing one alias of clang-tidy warning should silence other aliases

### ISSUE THREAD

**Issue by horenmar** [llvm/llvm-project#45204]
|  |  |
| --- | --- |
| Bugzilla Link | [45859](https://llvm.org/bz45859) |
| Version | unspecified |
| OS | All |
| CC | @EugeneZelenko,@adzenith |

## Extended Description 
I am using `NOLINT(cppcoreguidelines-pro-type-vararg)` in my code, but
`hicpp-vararg` warning is still reported by clang-tidy, even though it
is documented as alias of the former.

## Steps to reproduce

Run clang-tidy with "*-vararg" checks enabled on the following snippet
```cpp
void varargs(...) {}

void foo() {
    varargs(1, 2, 3); // NOLINT(hicpp-vararg)
}
```

The output will be 

```
/mnt/c/temp/Catch/1921.cpp:4:5: warning: do not call c-style vararg functions [cppcoreguidelines-pro-type-vararg]
    varargs(1, 2, 3); // NOLINT(hicpp-vararg)
    ^
Suppressed 1 warnings (1 NOLINT).
```

## Expected result

I expect no warnings to be issued, because `hicpp-vararg` and
`cppcoreguidelines-pro-type-vararg` warnings are documented to be aliases:

https://releases.llvm.org/9.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/hicpp-vararg.html

I would also prefer the output to report 1 suppressed warning, but I
think reporting 2 is also acceptable.

## Other information
This came up due to an issue in Catch2's issue tracker:
https://github.com/catchorg/Catch2/issues/1921

Catch2 uses `__builtin_constant_p` to trigger warnings in unevaluated
expressions, which clang-tidy flags as use of a variadic function (I
consider this as a bug, but will open separate report for it). When
suppressed with `NOLINT(cppcoreguidelines-pro-type-vararg)`, the warning
is still triggered when user enables hicpp-vararg check.

The linked issue confirms that this replicates for clang-tidy 7 and 9,
I also reproduced the issue on clang-tidy 10.

**Comment by Trass3r** [comment]
Duplicate #54319.
